src/main/scala/pl/mowczarek/love/actors/Field.scala

```diff
@@ -24,11 +33,25 @@ class Field(x: Int, y: Int, gameMap: ActorRef) extends Actor with ActorLogging {
       creatures.filter(_ != sender).foreach { creatureRef =>
         creatureRef forward command
       }
+
+    case command @ Migrate =>
+      (gameMap ? GetRandomField).mapTo[ActorRef].onComplete {
+        case Failure(ex) => throw ex
+        case Success(field: ActorRef) =>
+          creatures -= sender
+          log.info(s"Creature left field ($x, $y)")
+          field ! MigrateCreature(sender)
+      }
+
+    case MigrateCreature(creatureActor) =>
```

Comment:
It looks nice, but names of the commands are quite misleading. 'Migrate' command removes ref from list and 'MigrateCreature' adds it. 'Immigrate' and 'Emigrate' would be the proper ones I guess.